credoai/artifacts/data/tabular_data.py

```diff
@@ -52,16 +52,20 @@ def copy(self):
         return deepcopy(self)
 
     def _process_X(self, X):
-        """Standardize X data"""
+        """Standardize X data
+
+        Ensures X is a dataframe with string-named columns
+        """
         temp = pd.DataFrame(X)
         # Column names are converted to strings, to avoid mixed types
         temp.columns = temp.columns.astype("str")
         return temp
 
     def _process_y(self, y):
-        """Standardize y data"""
-        # if X is pandas object, and y is convertible, convert y to
-        # pandas object with X's index
+        """Standardize y data
+
+        If y is convertible, convert y to pandas object with X's index
+        """
         if isinstance(y, (np.ndarray, list)):
```

Comment:
What happens if this `isinstance()` call returns False? That would be the case if the user passed something entirely ill-formed --> seems like right now we're just assuming Y matches the docstring and is thus always a DF/Series, nd.Array, or List?